app/models/cohort_columns/unsheltered_days_homeless_last_three_years.rb

```diff
@@ -0,0 +1,23 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module CohortColumns
+  class UnshelteredDaysHomelessLastThreeYears < ReadOnly
+    attribute :column, String, lazy: true, default: :unsheltered_days_homeless
+    attribute :translation_key, String, lazy: true, default: 'Unsheltered Days Homeless in the last 3 years'
+    attribute :title, String, lazy: true, default: ->(model, _attr) { Translation.translate(model.translation_key) }
+    attribute :description_translation_key, String, lazy: true, default: 'Days in SO in the last 3 years with no overlapping sheltered dates ES, SH, TH, or PH after move-in date'
+    attribute :description, String, lazy: true, default: ->(model, _attr) { Translation.translate(model.description_translation_key) }
+
+    def cast_value(val)
+      val.to_i
+    end
+
+    def value(cohort_client)
+      GrdaWarehouse::Hud::Client.find_by(id: cohort_client.client_id).unsheltered_days_homeless_last_three_years
```

Comment:
This is going to cause an N+1.  You should be able to do:
```suggestion
      cohort_client.client.unsheltered_days_homeless_last_three_years
```

Reply:
However `unsheltered_days_homeless_last_three_years` is a massive calculation (additional N+1) so we'll probably need to add this to the `refresh_time_dependant_client_data` calculation in `cohort.rb`